libretro/libretro.c

```diff
@@ -2583,26 +2571,38 @@ void retro_init(void)
    if (environ_cb(RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY, &system_dir) && system_dir)
    {
       // if defined, use the system directory
-      retro_system_directory=system_dir;
+      strlcpy(
+            retro_system_directory,
+            system_dir,
+            sizeof(retro_system_directory));
    }
 
    const char *content_dir = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_CONTENT_DIRECTORY, &content_dir) && content_dir)
    {
       // if defined, use the system directory
-      retro_content_directory=content_dir;
+      strlcpy(
+            retro_content_directory,
+            content_dir,
+            sizeof(retro_content_directory));
    }
 
    const char *save_dir = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_SAVE_DIRECTORY, &save_dir) && save_dir)
    {
       // If save directory is defined use it, otherwise use system directory
-      retro_save_directory = *save_dir ? save_dir : retro_system_directory;
+      strlcpy(
+            retro_save_directory,
+            !string_is_empty(save_dir) ? save_dir : retro_system_directory,
```

Comment:
This form of ternary op check can look rather confusing. It's clearer to write:

    string_is_empty(save_dir) ? retro_system_directory : save_dir

Reply:
Good catch - I agree!

Reply:
Done!